### PR TITLE
feat: display presto and trino nested structures

### PR DESCRIPTION
**Pull request by froukees**
Solves #514

The information about subfields in nested structures is obtained from Presto&Trino, it's just not used and the current result isn't very readable.

This pull requests suggests:
- Leveraging cursor description in order to compose JSON with names of subfields for nested structures and storing it in the result CSV.
- Using a `ReactJson` component in order to display the JSON information in user-friendly fashion on the front-end.

### Query
```sql
SELECT
  CAST(
    ROW(
      1,
      'tp',
      ROW('xxx'),
      MAP(ARRAY [0.123456789], ARRAY [0.1]),
      timestamp '2012-10-31 01:00 UTC',
      INTERVAL '2' DAY,
      ARRAY ['aa', 'bb', 'cc'],
      IPADDRESS '10.0.0.1'
    ) AS ROW(
      id int,
      name varchar,
      nested ROW(x varchar),
      map map(decimal(38, 12), decimal(38, 5)),
      d TIMESTAMP WITH TIME ZONE,
      i INTERVAL DAY TO SECOND,
      arr ARRAY(varchar),
      ip IPADDRESS
    )
  ) AS struct
```

#### Light theme
image

#### Dark theme
image

#### Lush theme
image

#### New (auto-used) JSON transformer
image

#### Dealing with duplicate names of columns
Presto and Trino support duplicate names inside a `row()`

- For nested fields (`sub` in following example), the names has to be distinguished, otherwise they (all but one) are dropped on the Python worker as non-unique keys of a dictionary.
- For main columns, the same logic is applied in order to keep it consistent.

Proposed:
image

This can be definitely adjusted, e.g. different marking for the number of the (sub)column? Any preferences?

Double underscore:
image

Parenthesis:
image

### CODE REVIEW

**Review by czgu — COMMENTED** [review]
This is amazing, thanks for your work!

File: querybook/tests/test_lib/test_query_executor/test_clients/test_utils/test_presto_types.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                    )
+                ),
+            ),
+        ),
+    ],
+)
+def test_parse_from_string(string_definition: str, exp_type: PrestoType) -> None:
+    assert PrestoType.from_string(string_definition) == exp_type
+
+
```
[the remark this answers is not in the document]
Comment (froukees) [reply querybook/tests/test_lib/test_query_executor/test_clients/test_utils/test_presto_types.py:182]: sure, parameterized test was added bellow

File: querybook/webapp/lib/utils/boolean.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export function isBoolean(b: any): boolean {
+    if (typeof b === 'boolean') {
```
[the remark this answers is not in the document]
Comment (froukees) [reply querybook/webapp/lib/utils/boolean.ts]: thanks, much more concise

File: querybook/webapp/lib/query-result/transformer.tsx [hunk 3]
```diff
@@ -83,6 +85,53 @@ const queryResultTransformers: IColumnTransformer[] = [
...
+                }
+                return (
+                    <ReactJson
+                        collapsed={1} // keep first level expanded by default
+                        displayDataTypes={false}
+                        enableClipboard={false}
+                        name={false}
+                        quotesOnKeys={false}
+                        src={json}
+                        theme={{
```
[the remark this answers is not in the document]
Comment (froukees) [reply querybook/webapp/lib/query-result/transformer.tsx]: sure, moved

File: querybook/server/lib/query_executor/clients/trino.py [hunk 4]
```diff
@@ -58,24 +64,29 @@ def poll(self):
 
         return completed
 
-    def get_one_row(self):
-        return self._cursor.fetchone()
+    @staticmethod
+    def transform_row(row: List[Tuple], presto_types: List[PrestoType]) -> List[Any]:
```
[the remark this answers is not in the document]
Comment (czgu) [reply querybook/server/lib/query_executor/clients/trino.py]: yeah that naming makes sense

**Review by czgu — APPROVED** [review]

**Pull request merged**